kernel/kernel-util/src/test/java/org/sakaiproject/util/comparator/UserSortNameComparatorTest.java

```diff
@@ -68,4 +68,38 @@ public void underscoreCompare() {
         assertEquals(-1, comparator.compare(userB, userA));
         assertEquals(0, comparator.compare(userB, userB));
     }
+
+    @Test
+    public void namesWithSpacesCompare() {
+        UserSortNameComparator comparator = new UserSortNameComparator();
+        User userA = Mockito.mock(User.class);
+        when(userA.getSortName()).thenReturn("Dekfort, Apple");
+        User userB = Mockito.mock(User.class);
+        when(userB.getSortName()).thenReturn("Del Fintino, Pear");
+        User userC = Mockito.mock(User.class);
+        when(userC.getSortName()).thenReturn("Dekford", "Orange");
+        User userD = Mockito.mock(User.class);
+        when(userD.getSortName()).thenReturn("De'Leon", "Cactus");
+        User userE = Mockito.mock(User.class);
+        when(userE.getSortName()).thenReturn("Deleverde", "Mango");
+
+        assertEquals(-1, comparator.compare(userA, userB));
+        assertEquals(1, comparator.compare(userB, userA));
+        assertEquals(0, comparator.compare(userB, userB));
+        assertEquals(-1, comparator.compare(userD, userC));
+        assertEquals(-1, comparator.compare(userD, userE));
+    }
+
+    @Test
+    public void SAK21743() {
```

Comment:
avoid naming methods after JIRA issues and instead use a comment
i.e. fullnameWithSpaces()

Reply:
feel free to edit and rename. branch is open to editing.